### PR TITLE
One love for PNMixer

### PR DESCRIPTION
**Pull request by elboulangero**
- Two minor issues fixed (#146 #149)
- A big deal of maintenance, mainly for translation: move away from `intltool` and use a modern version of `gettext` instead.

The two main reasons for moving away from intltool:

- intltool is a wrapper on top of gettext, which purpose is to overcome limitations in gettext. However, modern gettext don't have these limitations, so intltool is not relevant anymore.
- using gettext, the workflow is a bit different, and we now version the potfile. And this is good ! Because we need to give the potfile to the TP for translation when we release. The way to do that before was to create an archive with make dist, which included the pot file, and add it to GitHub on the release page. Now that the potfile is versioned, a simple tag is enough, good-bye `make dist`.

### CODE REVIEW

**Comment by elboulangero** [comment]
The Ubuntu build fails because it lacks `autopoint`. Autopoint is a new build dependency related to gettext.

So I think the only thing needed here is to add autopoint to the dockerfiles. On Debian and Ubuntu, the package is just named `autopoint`, so that's easy.

**Comment by elboulangero** [comment]
And btw, intltool can be removed from the dockerfiles.

**Comment by hasufell** [comment]
> So I think the only thing needed here is to add autopoint to the dockerfiles.

I've been considering to rework the travis thing anyway, since currently I have to manually maintain the images, more or less at least, which is a nuisance.

File: README.md [hunk 1]
```diff
@@ -74,6 +74,8 @@ manually if your distribution does not provide a package.
 
 First, make sure you have the required __dependencies__:
 - build:
+	- autopoint (for bootstrapping)
+	- autoreconf (for bootstrapping)
```
Comment (hasufell) [inline README.md]: autoreconf is just a wrapper, so not really required and afais not used by `autogen.sh` either

File: README.md [hunk 2]
```diff
@@ -74,6 +74,8 @@ manually if your distribution does not provide a package.
 
 First, make sure you have the required __dependencies__:
 - build:
+	- autopoint (for bootstrapping)
```
Comment (hasufell) [inline README.md]: autopoint is part of gettext, which is already listed. Or are there distros that ship this separately?

Comment (elboulangero) [inline README.md]: Yes it's a separate package on Debian

File: README.md (hunk 1 above)
Comment (elboulangero) [inline README.md]: Well I think nowadays there's no need to carry around a massive autogen script, autoreconf do the job as well. That's how it does in GLib and GTK+, and every distro provide these libraries, right ? So I think it's safe to follow their example.

Plus, there are our main dependencies, so if distros can compile GLib and GTK+, then they can

Comment (hasufell) [inline README.md]: On some distros, this dep is called [sys-devel/autoconf-wrapper](https://github.com/gentoo/gentoo/tree/master/sys-devel/autoconf-wrapper).

File: README.md (hunk 2 above)
Comment (hasufell) [inline README.md]: Which distro do we follow then for listing the dependencies? I feel like this is becoming a wild mix. Maybe make separate sections for different distros?

Comment (hasufell) [inline README.md]: IMO, I think we should stick to upstream. Autopoint is part of gettext: http://git.savannah.gnu.org/cgit/gettext.git/tree/gettext-tools/misc/autopoint.in
and autoreconf is part of autoconf: http://git.savannah.gnu.org/cgit/autoconf.git/tree/bin/autoreconf.in

Both are already listed. Distro-specific notes should be added separately.

Comment (elboulangero) [inline README.md]: Yep maybe we should remain simple. Developers can handle these build dependencies by themselves, these are just details.

Comment (elboulangero) [inline README.md]: I didn't know it was such a mess :)

File: src/audio.c [hunk 3]
```diff
@@ -368,6 +369,23 @@ audio_get_channel(Audio *audio)
 }
 
 /**
+ * Whether the card has mute capabilities.
+ *
+ * @param audio an Audio instance.
+ * @return TRUE if the card can be muted, FALSE otherwise.
+ */
+gboolean
+audio_has_mute(Audio *audio)
```
Comment (hasufell) [inline src/audio.c]: is it possible that `*audio` is NULL? In that case, we'd get NULL dereference in line 380

Comment (elboulangero) [inline src/audio.c]: It's not possible, the audio instance is created and destroyed in the main function. It lives during the whole application runtime. That's why there's no check for `(audio == NULL)` in the whole `audio.c` file.

**Comment by elboulangero** [comment]
Any news on this ? I'm kind of waiting to merge that, rework details here and there, and tag a release.

**Comment by elboulangero** [comment]
Alright, merged !

**Pull request merged**